app/src/features/MovieDetailsScreen/index.tsx

```diff
@@ -0,0 +1,221 @@
+import { Fragment, useCallback, useEffect, useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { Col, Row } from 'react-bootstrap';
+import { Backdrop, CircularProgress } from '@mui/material';
+import useAsyncError from 'hooks/useAsyncError';
+import { getMovie } from 'redux/movie/reducers/Movie';
+import { getScreeningsForMovie } from 'redux/movie/reducers/Screenings';
+import * as selectors from 'redux/movie/selectors';
+import Movie from 'models/Movie';
+import Screening from 'models/Screening';
+import { removeTime } from 'util/dateUtils';
+import ScheduleItem from 'models/Screening/ScreeningSchedule';
+import ScreeningSchedule from 'components/ScreeningSchedule';
+import SomethingWentWrong from 'components/SomethingWentWrong';
+import * as S from './index.styles';
+import EmbeddedVideo from 'components/EmbeddedVideo';
+
+const MovieDetailsScreen = (): JSX.Element => {
+    const { id } = useParams();
+    // This will never happen because the route wouldn't match
+    // but typescript requires the null check
+    if (id === undefined) {
+        return <SomethingWentWrong></SomethingWentWrong>;
+    }
+
+    const throwAsyncError = useAsyncError();
+    const dispatch = useDispatch();
+
+    const movie = useSelector(selectors.getMovie);
+    const movieStatus = useSelector(selectors.getMovieStatus);
+
+    const movieScreenings = useSelector(selectors.getScreeningsForMovie);
+    const movieScreeningsStatus = useSelector(
+        selectors.getScreeningsForMovieStatus
+    );
+
+    const isMovieLoading = movieStatus === 'pending' || movieStatus === 'idle';
+    const areScreeningsLoading =
+        movieScreeningsStatus === 'pending' || movieScreeningsStatus === 'idle';
+
+    const screeningGroups = useMemo(() => {
+        // Filter out screenings that have ended
+        // let filteredScreenings = movieScreenings.filter(
+        //     (s) => s.movieStart.getTime() >= Date.now()
+        // );
+        let filteredScreenings = movieScreenings;
+
+        // group screening by date
+        // Dictionary<date: string, screenings: Screenings[]>
+        let dict = filteredScreenings.reduce(
```

Comment:
Zar ne vraca reduce novi niz? U tom slucaju dal ti treba filteredScreenings promenljiva?

Reply:
Reduce ce vratiti novi objekat u ovom slucaju. Ideja filteredScreenings je da pre reduce izbacim screening-e koji su vec prosli ali mislim da je to zakomentarisano jer je lakse testirati ako ne moras da dodajes novi screening svaki dan

Reply:
ok, nevezano od toga, moze li dict da bude const?